sdk/core/core-amqp/CHANGELOG.md

```diff
@@ -2,6 +2,11 @@
 
 ## 2.0.0 (Unreleased)
 
+### Breaking changes
+
+- `DefaultDataTransformer` is no longer exported.
```

Comment:
Not exporting the DefaultDataTransformer, but continuing to export DataTransformer interface does not give us much. The main reason for us to do this is to keep the option open in the future to implement the "serializer" in any other way.

I would recommend removing the `dataTransformer` property from `ConnectionContextBase` to allow us to stop exporting this interface from core-amqp. This will result in copying the code in the DefaultDataTransformer to Service Bus and EventHubs and use it like a static class instead of using it from the context.

In the future when we have the common code between SB & EH without needing to expose it users from core-amqp, we can move this